lib/fhir/resources/value-set.js

```diff
@@ -0,0 +1,47 @@
+/**
+* Copyright (c) 2017-present, Jembi Health Systems NPC.
+* All rights reserved.
+*
+* This source code is licensed under the BSD-style license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict'
+
+const constants = require('../../constants')
+const FhirCommon = require('../common')
+const QueryUtils = require('../query-utils')
+
+module.exports = (mongo) => {
+  const fhirCommon = FhirCommon(mongo)
+  const queryUtils = QueryUtils(mongo)
+
+  return {
+    name: 'ValueSet',
+
+    searchFilters: (ctx, callback) => {
+      const queryParams = ctx.query
+      const supportedParams = {
+        url: {}
+      }
+
+      queryUtils.validateAndParseQueryParams(queryParams, supportedParams, (badRequest, queryObject) => {
+        if (badRequest) {
+          return callback(null, fhirCommon.buildHTTPOutcome(400, 'error', 'invalid', badRequest))
+        }
+
+        let query = { $and: [] }
+
+        if (queryObject['url']) {
+          query['$and'].push({ 'codeSystem.system': queryObject['url'][constants.NO_MODIFER] })
```

Comment:
The "url" param needs to point to the "ValueSet.url" property.
We will need to add another supportedParam for "system" which targets the property we are using here.